File: Python/boj/Silver/9095.py
import sys

if __name__ == "__main__":
    try:
        sys.stdin = open("input.txt", "r")
    except FileNotFoundError:
        pass

    T = int(input())
    testcase = [int(input()) for _ in range(T)]
    result = []

    for num in testcase:
        dp = [0] * (num + 1)

        for i in range(1, num + 1):
            if i == 1:
                dp[i] = 1
            elif i == 2:
                dp[i] = 2
            elif i == 3:
                dp[i] = 4
            else:
                dp[i] += dp[i - 1] + dp[i - 2] + dp[i - 3]

        result.append(dp[num])

    print("\n".join(map(str, result)))

File: JavaScript/boj/Silver/9095.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');

const n = +input[0];
const dp = Array.from({ length: n + 1 }, () => 0);
const result = [];

dp[1] = 1;
dp[2] = 2;
dp[3] = 4;

for (let i = 4; i <= 11; i++) {
  dp[i] = dp[i - 1] + dp[i - 2] + dp[i - 3];
}

for (let i = 1; i < input.length; i++) {
  result.push(dp[input[i]]);
}

console.log(result.join('\n'));
